### ISSUE TITLE
Powershell module not importing or cant find

### ISSUE THREAD

**Issue by pr1malc0de** [dotnet/interactive#1720]
### Describe the bug
Importing a module doesnt seem to to 

### Please complete the following:

**Which version of .NET Interactive are you using?** There are a few ways to find this out:

1.0.255901

- OS
    - [x ] Windows 10
    - [ ] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [ ] Chrome
    - [x ] Edge
    - [ ] Firefox
    - [ ] Safari
 - Frontend
    - [ ] Jupyter Notebook
    - [ ] Jupyter Lab
    - [ ] nteract
    - [ ] Visual Studio Code 
    - [x ] Visual Studio Code Insiders
    - [ ] Other (please specify)

### Screenshots
![image](https://user-images.githubusercontent.com/662108/140942423-a256e45f-c018-4419-afa3-202c2d190d1e.png)

**Comment by pr1malc0de** [comment]
Anybody have any idea if this should even be possible?

**Comment by jonsequitur** [comment]
@daxian-dbw Any ideas here?

**Comment by daxian-dbw** [comment]
@pr1malc0de Can you follow the instruction in the error message and see what you get when running `Import-Module RemoteDesktop`?

**Comment by pr1malc0de** [comment]
@daxian-dbw hi! I ran the import command it seemed ok (see screenshot)

**Comment by pr1malc0de** [comment]
@daxian-dbw anything else i can try?

**Comment by Szeraax** [comment]
I think the problem is that the RemoteDesktop module isn't built for Pwsh and I don't think that there is anything that .Net interactive can do to fix that. 

If you look at the number of modules in each of my $PSModulePath folders, you'll see that I have 106 modules listed in the SAME PATH on WindowsPowershell 5.1, but only 64 modules on Pwsh 7.2. See image:
![image](https://user-images.githubusercontent.com/6242511/145163788-9c387c87-3bb0-463d-871c-193628a33806.png)


If you open up a pwsh shell and run `Import-Module RemoteDesktop -SkipEditionCheck`, you'll see that Pwsh CANNOT open this module without doing a New-PSSession over to a WindowPowershell session and then import the commands into the current session. 

I don't know if .net Interactive would want to add in the complexity of trying to support remote sessions to WindowsPowershell, however, if @jonsequitur knows whether they have an upcoming feature for supporting remote sessions over powershell remoting (don't know if SSH remoting could even be possible too), then this would be trivial (or automatically backported) as a part of that feature. 

The fact that the module does import at all suggests that there MAY be hope yet. 

There are various other modules that don't load in Pwsh when doing -SkipEditionCheck like Appx, PSWorkflow, and Servermanager. I wouldn't hold my breath.

**Comment by daxian-dbw** [comment]
@pr1malc0de Which `RemoteDesktop` module were you using? [The one on PowerShell gallery](https://www.powershellgallery.com/packages/RemoteDesktop/0.0.0.2) is broken and it doesn't look like what you were using.

**Comment by Szeraax** [comment]
@daxian-dbw I feel confident that he's talking about the one included by default with windows:

![image](https://user-images.githubusercontent.com/6242511/145652102-4e4dde23-5c7d-424a-9f9c-cc2d7c092aad.png)

**Comment by pr1malc0de** [comment]
@Szeraax @daxian-dbw  thats correct it's the one that is included in Rsat tools I believe. I had thought that using the switch -UseWindowsPowerShell (even though my screenshot was before I tried) would just work, didn't realise that there would need to be compatibility with the .net interactive

**Comment by daxian-dbw** [comment]
@pr1malc0de @Szeraax Thank you both for the clarification! It turns out `import-Module RemoteDesktop -UseWindowsPowerShell` just silently failed because PowerShell remoting depends on `BinaryFormatter` even though it's obsolete, and due to the settings in .NET interactive, using `BinaryFormatter` for serialization/deserialization throws `NotSupportedException`. Below is an example with importing `PersistentMemory` with `-UseWindowsPowerShell`:

![image](https://user-images.githubusercontent.com/127450/145659009-f3275180-0dca-4719-aba7-019c3ccbbc40.png)

@jonsequitur somehow the using of `BinaryFormatter.Serialize/Deserialize` is disallowed at run time. Would it be possible to enforce the obsoletion at compilation time, while allow it at runtime?

**Comment by jonsequitur** [comment]
Using `BinaryFormatter` directly should already emit a warning a compile time. For security reasons, we don't plan to allow its usage at runtime.

More details here: https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/binaryformatter-serialization-obsolete

**Comment by daxian-dbw** [comment]
I know that, but `BinaryFormatter` is used in PowerShell remoting and for back compatibility reason (remoting to Windows PowerShell 5.1), we had to keep it and [suppress the warning](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/remoting/common/WireDataFormat/RemoteSessionCapability.cs#L110-L112) in PowerShell. I just post a message to my team to re-evaluate this dependency.

If re-enabling `BinaryFormatter` functionality won't be possible in .NET interactive, then let's consider this issue by-design. Also, PowerShell remoting won't work in .NET interactive either due to the same reason.

**Comment by daxian-dbw** [comment]
Here is the response from our remoting/security SME:
> Yes, we are aware of the issues using BinaryFormatter, and determined that there is no security hole since we control the data we serialize/deserialze.  The information is used by Exchange and so we've kept it for remoting backwards compatibility.
> We can change the code to not use the serializer, but doing so may break compatibility with older WinRM endpoints, such as Exchange.
> Ok I just looked closely at the code.  We have already removed the `TimeZone` deserialization, which is where there could be a security concern.  So what we have left represents no security concern.  If we decide to remove `TimeZone` altogether, we will likely break compatibility with Exchange endpoints.  We would also be in violation of the PSRP.

@jonsequitur Given that there is no security concern with the use of `BinaryFormatter` in PowerShell remoting, would you consider enable the use of it at run time for .NET Interactive? You can continue to disallow the direct use of it in the .NET Interactive project source code.

**Comment by pr1malc0de** [comment]
@jonsequitur any update on this do you think it will be possible now? Cheers

**Comment by BrianTJackett** [comment]
Run into (similar) issue when try to load the Microsoft.Graph.Authentication module in the OnLoadAsync method of custom extension.  Able to load this module directly into a PowerShell Polyglot Notebook code block as a workaround, but would prefer to be able to load it dynamically / automatically in the background if possible.

**Comment by BrianTJackett** [comment]
I see PowerShell v7.3.0 just released within last month.  Do you know timeline for 7.3.2 to release?  Will use workaround of manually loading dependent PowerShell modules for the time being, but plan to revisit this in the future.

**Comment by daxian-dbw** [comment]
7.3.1 will be released in December. 7.3.2 will likely be in January at the earliest.

**Comment by BrianTJackett** [comment]
It looks like the next version of PowerShell SDK for Microsoft Graph (internal preview build at the moment) is able to load in .Net Interactive.  Waiting on upcoming public preview release of that and PowerShell v7.3.2 before moving forward.  Appreciate your insights so far @daxian-dbw.
